### PR TITLE
fix: unresponsive ledger device

### PR DESCRIPTION
**Pull request by r4mmer**
### Summary

Fix the issues described in #473

### Acceptance Criteria
- Update ledger transport dependency
- Update hathor wallet lib dependency
- Do not redirect to locked screen when starting a hardware wallet
- During preload, clean any previous hardware wallet access data.
- Add better logs to debug issues with ledger communication

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: public/preload.js [hunk 1]
```diff
@@ -19,6 +19,17 @@ process.once('loaded', () => {
   // Set closed in localStorage, so user does not open in the wallet page
   localStorage.setItem('localstorage:closed', 'true');
 
+  const accessDataRaw = localStorage.getItem('localstorage:accessdata');
+  if (accessDataRaw) {
+    // check if the access data is from a hardware wallet, if so we need to remove it
+    // This ensures that a previously closed hardware wallet will not be loaded again
+    // unless the device is connected
+    const accessData = JSON.parse(accessDataRaw);
+    if ((accessData?.walletFlags & 0x02) > 0) {
```
Comment (r4mmer) [inline public/preload.js:28]: `0x02` is the hardware wallet flag ([ref](https://github.com/HathorNetwork/hathor-wallet-lib/blob/v1.0.4/src/types.ts#L204)).
We can expose `WALLET_FLAGS` from the lib on the next version and import it here but I wish to release this fix asap so I hardcoded the value here for now.

File: package.json [hunk 2]
```diff
@@ -25,8 +25,8 @@
   "version": "0.27.1-rc2",
   "private": true,
   "dependencies": {
-    "@hathor/wallet-lib": "1.0.2",
-    "@ledgerhq/hw-transport-node-hid": "6.27.1",
+    "@hathor/wallet-lib": "1.0.4",
+    "@ledgerhq/hw-transport-node-hid": "^6.28",
```
Comment (tuliomir) [inline package.json:29]: question: Could we use a pinned version for this dependency? This has been the default since #422

File: src/storage.js [hunk 3]
```diff
@@ -198,14 +198,17 @@ export class LocalStorageStore {
     }
   }
 
-  cleanWallet() {
-    this.removeItem(IS_HARDWARE_KEY);
-    this.removeItem(CLOSED_KEY);
+  cleanAccessData() {
```
Comment (tuliomir) [inline src/storage.js:201]: suggestion: Could you add descriptions for these methods? It's not clear to me when we should use `clearWallet` or `cleanAccessData`.

Comment (r4mmer) [reply src/storage.js:204]: Done

File: package.json (hunk 2 above)
Comment (r4mmer) [reply package.json]: Done

File: public/ledger.js [hunk 4]
```diff
@@ -134,7 +136,10 @@ class Ledger {
           }
         }
       },
-      error: error => {},
+      error: error => {
+        console.log('Error communicating with device');
+        console.log(error);
```
Comment (tuliomir) [inline public/ledger.js:141]: nitpick: Maybe switch these error messages to `console.error` instead?

**Review by tuliomir — APPROVED** [review]

File: src/screens/StartHardwareWallet.js [hunk 5]
```diff
@@ -128,9 +129,6 @@ class StartHardwareWallet extends React.Component {
         xpub,
         hardware: true,
       });
-      LOCAL_STORE.markBackupDone();
-
-      this.props.history.push('/wallet/');
```
Comment (pedroferreira1) [inline src/screens/StartHardwareWallet.js:133]: After starting the wallet we must change to the wallet screen. How is it done now? Through the sagas?

File: package.json [hunk 6]
```diff
@@ -25,8 +25,8 @@
   "version": "0.27.1-rc2",
   "private": true,
   "dependencies": {
-    "@hathor/wallet-lib": "1.0.2",
-    "@ledgerhq/hw-transport-node-hid": "6.27.1",
+    "@hathor/wallet-lib": "1.0.4",
```
Comment (pedroferreira1) [inline package.json:28]: Why do we need this upgrade?

File: public/preload.js (hunk 1 above)
Comment (pedroferreira1) [reply public/preload.js:28]: Can you create the issue for that, please?

File: public/ledger.js [hunk 7]
```diff
@@ -175,11 +180,21 @@ class Ledger {
...
-      .then((response) => { resolve(response) }, (error) => { reject(error) })
+      .then(transport => {
+        console.log(`Sending to ledger ${command} ${p1} ${p2}`);
+        return transport.send(ledgerCLA, command, p1, p2, data);
+      })
+      .then((response) => {
+        console.log(`Got response ${response.toString('hex')}`);
+        resolve(response);
+      }, (error) => {
+          console.log(error);
```
Comment (pedroferreira1) [inline public/ledger.js:191]: Is this indentation correct?

File: public/ledger.js [hunk 8]
```diff
@@ -240,7 +255,9 @@ class Ledger {
         this.transport.decorateAppAPIMethods(this, this.methods, "HTR");
         resolve(transport);
       }, (e) => {
-        reject(Ledger.parseLedgerError(e));
+          console.log('Error creating transport');
```
Comment (pedroferreira1) [inline public/ledger.js:258]: Is this indentation correct?

File: src/storage.js [hunk 9]
```diff
@@ -198,14 +198,26 @@ export class LocalStorageStore {
     }
   }
 
-  cleanWallet() {
-    this.removeItem(IS_HARDWARE_KEY);
-    this.removeItem(CLOSED_KEY);
+  /**
+   * Clean access data from persistent storage, making the wallet 'not loaded'
+   */
+  cleanAccessData() {
```
Comment (pedroferreira1) [inline src/storage.js:204]: Why did you need to change this?

File: src/screens/StartHardwareWallet.js (hunk 5 above)
Comment (r4mmer) [reply src/screens/StartHardwareWallet.js:133]: This was here before to avoid changing the screen too fast (i.e. flashing the scren) before showing the loading addresses screen.
This has been fixed and is no longer needed, the saga will redirect to the loading addresses screen when required.

File: package.json (hunk 6 above)
Comment (r4mmer) [reply package.json:28]: It's not necessary, Its just that wallet-lib v1.0.3 fixed an issue with the typing of the wallet-lib, which makes my LSP more useful (autocompletes, documentation, method signatures, etc.).

File: src/storage.js (hunk 9 above)
Comment (r4mmer) [reply src/storage.js:204]: When starting a hardware wallet we were having an issue that the acess data was being deleted during the `startWallet` saga, this would fail the process to start a wallet and if we clicked on retry it would work.

The issue was that we were being redirected to the locked screen as a hardware wallet, and we call `cleanWallet` to show the `wallet_type` screen.
This was fixed by not redirecting the hardware wallet to the locked screen during the `startWallet` saga and not cleaning the access data on `cleanWallet`, this is why I created the `cleanAccessData` if we really want to delete the access data from the localStorage.

File: public/preload.js (hunk 1 above)
Comment (r4mmer) [reply public/preload.js:28]: Done in https://github.com/HathorNetwork/hathor-wallet-lib/issues/601

File: public/ledger.js (hunk 8 above)
Comment (r4mmer) [reply public/ledger.js:258]: Fixed

File: public/ledger.js (hunk 7 above)
Comment (r4mmer) [reply public/ledger.js:191]: Fixed

**Review by pedroferreira1 — APPROVED** [review]
